### ISSUE TITLE
`list_remove_unrelated_type` and `iterable_contains_unrelated_type` do not report on implicit receiver calls in an extension method

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#3702]
**Describe the issue**

`list_remove_unrelated_type` and `iterable_contains_unrelated_type` do not report on implicit receiver calls in an extension method.

**To Reproduce**

```dart
extension on List<int> {
  void f() {
    remove('string'); // LINT
    this.remove('string'); // LINT
    remove(1); // OK
  }
}
```

**Expected behavior**

The lines with LINT should be reported as lint.
